src/audio/multiband_drc/multiband_drc.h

```diff
@@ -8,14 +8,14 @@
 #define __SOF_AUDIO_MULTIBAND_DRC_MULTIBAND_DRC_H__
 
 #include <sof/audio/module_adapter/module/generic.h>
-#include <sof/audio/crossover/crossover.h>
 #include <sof/audio/drc/drc.h>
 #include <sof/math/iir_df2t.h>
 #include <sof/audio/component.h>
 #include <sof/audio/data_blob.h>
 #include <sof/platform.h>
 #include <stdint.h>
 #include "user/multiband_drc.h"
+#include "../crossover/crossover.h"
```

Comment:
@marc-hb , how about add one more path to zephyr/CMakeList with below diff:
--- a/zephyr/CMakeLists.txt
+++ b/zephyr/CMakeLists.txt
@@ -90,6 +90,7 @@ ExternalProject_Add(sof_logger_ep
 # default SOF includes
 target_include_directories(SOF INTERFACE ${RIMAGE_TOP}/src/include)
 target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/include)
+target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/audio)
 target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/arch/${ARCH}/include)
 target_include_directories(SOF INTERFACE ${sof_top_dir}/third_party/include)


then all these can be changed with <crossover/crossover.h> ?

Reply:
@marc-hb we do! I think ".." is better in this case to make it very clear this module is reusing definitions of a peer module (see my comments to https://github.com/thesofproject/sof/pull/8593 ).

Reply:
>  how about add one more path to zephyr/CMakeList with below diff:

(I edited your diff, edit it again to see how)

Yes please, something like that but probably with some `CMakeLists.txt:if(CONFIG_...)`. It would be interesting to check whether that solves the disassembly difference between Linux and Windows. That can be done with test draft PR (no reviewers / no github spam)



No no such disassembly difference in https://github.com/thesofproject/sof/actions/runs/7137933378/job/19438902490?pr=8593 even though it has one `#include "../"`

Reply:
> to make it very clear this module is reusing definitions of a peer module

There is no requirement for all `target_include_directories()` to be in the top-level makefile.

Reply:
@kv2019i @marc-hb , yes we need a conclusion for the cross module header usage:
1. use "../" or "../../"
2. add new target_include_directories with +target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/audio)

Reply:
@johnylin76 @btian1 I have in my improvement ideas list move of crossover filter bank to math library to avoid the tie between DRC and crossover. Need to do it latest when we next need other recursive filter bank. So this include would become include from math headers.

@btian1 But I don't think it fits this PR scope, should be done after since it may need small changes to algorithms code.  If you can test our option 2. it would be nice.

Reply:
@marc-hb @btian1 @singalsu To me, adding a non-include folder into target_include_directories is opening a can of worms. Then the build is searching for headers in the implementation parts of the code base and this can lead to surprises. I personally prefer the explicit "../../foo.h" includes, to make it very explicit something unusual is happening here, and only specific foo.h is included. This really only affects modules that are directly reusing portions of other peer modules.

Reply:
Thanks for thoughts @kv2019i ! Makes sense. I hope we can soon clean up this dependency into maths library. It's been causing difficulty with testbench build (shared libraries time).

Reply:
> I think ".." is better in this case to make it very clear this module is reusing definitions of a peer module

I tested this and it Just Works:
```
zephyr_library_include_directories_ifdef(CONFIG_COMP_MULTIBAND_DRC
  ${SOF_AUDIO_PATH}/crossover
)
```

This expresses the multiband->crossover dependency (or any other similar dependency) very clearly and in a regular, "CMake-native" way.


> To me, adding a non-include folder into target_include_directories is opening a can of worms. 
> Then the build is searching for headers in the implementation parts of the code base and this can lead to surprises.

I don't see what could go wrong: there are very few .h files in the .c directories after all.

In fact the "real" question is: why is the .h file of an _external module API_ located in the same directory as some .c files? This here is really the root-cause/of all evil.

If a "peer module" wants to be re-usable then I'm sure it can afford some `include/` subdirectory, that's really not vey demanding and it clarifies what .h is external versus internal. Then:

```
zephyr_library_include_directories_ifdef(CONFIG_COMP_MULTIBAND_DRC
  ${SOF_AUDIO_PATH}/crossover/include
)
```

and job done.